openslides_backend/action/actions/user/user_mixin.py

```diff
@@ -19,4 +24,158 @@ def update_instance(self, instance: Dict[str, Any]) -> Dict[str, Any]:
+                raise ActionException(
+                    f"User {user_fqid.id} cannot delegate his vote, because there are votes delegated to him."
+                )
+            mapped_field = f"vote_delegated_${meeting_id}_to_id"
+            user_delegated_to = self.datastore.fetch_model(
+                FullQualifiedId(Collection("user"), delegated_to_id),
+                [mapped_field],
+                lock_result=True,
+            )
+            if user_delegated_to.get(mapped_field):
+                raise ActionException(
+                    f"User {user_fqid.id} cannot delegate his vote to user {delegated_to_id}, because that user has delegated his vote himself."
+                )
+
+    def check_vote_delegations__from_ids(
+        self, instance: Dict[str, Any], user_fqid: FullQualifiedId
+    ) -> None:
+        mapped_fields = [
+            f"vote_delegated_${meeting_id}_to_id"
+            for meeting_id, delegated_from in instance[
+                "vote_delegations_$_from_ids"
+            ].items()
+            if delegated_from
+        ]
+        if not mapped_fields:
+            return
+        user_self = self.datastore.fetch_model(
+            user_fqid, mapped_fields, lock_result=True, exception=False
+        )
+        if "vote_delegated_$_to_id" in instance:
+            update_dict = {
+                f"vote_delegated_${meeting_id}_to_id": delegated_to
+                for meeting_id, delegated_to in instance[
+                    "vote_delegated_$_to_id"
+                ].items()
+            }
+            user_self.update(update_dict)
+        for meeting_id, delegated_from_ids in instance[
+            "vote_delegations_$_from_ids"
+        ].items():
+            if user_fqid.id in delegated_from_ids:
+                raise ActionException(
+                    f"User {user_fqid.id} can't delegate the vote to himself."
+                )
+            if user_self.get(f"vote_delegated_${meeting_id}_to_id"):
+                raise ActionException(
+                    f"User {user_fqid.id} cannot receive vote delegations, because he delegated his own vote."
+                )
+            mapped_field = f"vote_delegations_${meeting_id}_from_ids"
+            error_user_ids: List[int] = []
+            for user_id in delegated_from_ids:
+                user = self.datastore.fetch_model(
+                    FullQualifiedId(Collection("user"), user_id),
+                    [mapped_field],
+                    lock_result=True,
+                )
+                if user.get(mapped_field):
+                    error_user_ids.append(user_id)
+            if error_user_ids:
+                raise ActionException(
+                    f"User(s) {error_user_ids} can't delegate their votes , because they receive vote delegations."
+                )
+
+    def check_existence_of_to_and_from_users(self, instance: Dict[str, Any]) -> None:
```

Comment:
I _think_ this method is unecessary because the relation handling should throw an error if a related model does not exist. Did you test without it?

Reply:
To check for that error, the method is not necessary. The method exist, because of the code from [here](https://github.com/OpenSlides/openslides-backend/blob/e55d7804507859e49f9bd4eb62860336c03b2f57/openslides_backend/action/actions/user/temporary_user_mixin.py#L36). It gives the error message more specific ("The following users were not found: {...}"). Without this method it would be "The following models do not belong to meeting 222: ['user/7']", because it would be detected in asserts_belong to meeting.py. I think there has been an issue to make the error message more understandable.

Reply:
Ah okay, I see. Since we have it now, we may just keep it, although I would say the latter error messsage would also suffice